extensions/amp-analytics/0.1/analytics-root.js

```diff
@@ -287,18 +312,72 @@ export class AnalyticsRoot {
    * @param {string} selector DOM query selector.
    * @param {?string=} selectionMethod Allowed values are `null`,
    *   `'closest'` and `'scope'`.
-   * @param {boolean=} opt_multiSelectorOn multi-selector expriment
    * @return {!Promise<!AmpElement>} AMP element corresponding to the selector if found.
    */
-  getAmpElement(context, selector, selectionMethod, opt_multiSelectorOn) {
+  getAmpElement(context, selector, selectionMethod) {
     return this.getElement(context, selector, selectionMethod).then(element => {
+      this.verifyAmpElements_([element], selector);
+      return element;
+    });
+  }
+
+  /**
+   * Searches for the AMP element(s) that matches the selector
+   * within the scope of the analytics root in relationship to
+   * the specified context node.
+   *
+   * @param {!Element} context
+   * @param {string} selector DOM query selector.
+   * @param {?string=} selectionMethod Allowed values are `null`,
+   *   `'closest'` and `'scope'`.
+   * @param {boolean=} opt_multiSelectorOn multi-selector expriment
+   * @param {Array} opt_elementsReference reference to unique elements
+   * @return {!Promise<!Array<!AmpElement>>} Array of AMP elements corresponding to the selector if found.
+   */
+  getAmpElementOrElements(
```

Comment:
I have an idea, let me know what do you think. We create a function
`getElements(context, selectors, selectionMethod)`.
The this function will always return an array of unique element selected.
And then in the `event.js`, we would call
```
getElements().then(elements => {
 for (let i=0; i<elements.length; i++) {
   registerForElement_(element[i])
 }
})
```

Reply:
Hmm, I tired doing that version, however, we when register the element's listener, the `getReadyPromise()` requires the element and the selector. So `getElements()` would be more complicated (retaining unique elements while returning some sort of mapping between selector and elements). 

Do you still think it's worth to go down that route?

Reply:
Talked offline and decided to change `getReadyPromise()` (since we will need to change it in the future too) to support the `getElements()` route.